payload/Library/Application Support/nudge/Resources/nudge

```diff
@@ -438,8 +439,15 @@ def main():
     if LooseVersion(minimum_os_version_major) > get_os_version_major():
         # This is a major upgrade now and needs the app. We shouldn't
         # perform minor updates.
-        if not os.path.exists(PATH_TO_APP):
-            print 'Update application not found! Exiting...'
+        if LOCAL_URL_FOR_UPGRADE:
+            # Reassign the global PATH_TO_APP with the specified local
+            # upgrade URL
+            PATH_TO_APP = LOCAL_URL_FOR_UPGRADE
+        if not os.path.exists(PATH_TO_APP) and not LOCAL_URL_FOR_UPGRADE:
```

Comment:
Is there potentially an edge case here? If you are overriding PATH_TO_APP, but then here you check if that path exists, is it actually going to exist?

IOW, I think these if statements should be flipped. Check for PATH_TO_APP first to see if the .app is there, then if it is but you override it with local_url_for_upgrade, it will be overridden.

It could be identical, but I believe there might be a tiny edge case we are missing here and I want to be safer around the original behavior of the code.

Reply:
Specifically talking about L446

Reply:
Perhaps a more explicit flow?

```suggestion
        if LOCAL_URL_FOR_UPGRADE:
            # Reassign the global PATH_TO_APP with the specified local
            # upgrade URL
            PATH_TO_APP = LOCAL_URL_FOR_UPGRADE
        else:
            if not os.path.exists(PATH_TO_APP)
                print ('Update application not found! Exiting...')
                exit(1)
```

This way we're only checking the path if we actually need to. `LOCAL_URL_FOR_UPGRADE` will be false given the default `False` assignment on L398 unless the admin desires to override the default nudge behavior, and only in that case do we care if the app is present.

Thoughts?

Reply:
Yeah that sounds good.